akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToResponseMarshallers.scala

```diff
@@ -118,7 +126,14 @@ trait LowPriorityToResponseMarshallerImplicits {
                   case Marshalling.WithOpenCharset(`bestMT`, marshal)    ⇒ () ⇒ marshal(bestCS)
                 }
             }).toList
+
+            // TODO we could either special case for certrain known types,
+            // or extend the entity support to be more advanced such that it would negotiate the element content type it
```

Comment:
Not really related, notice that content type from Accept was already selected. 
This is only for marshaller of a single element that builds up into the stream that together is of type of that ContentType.

I.e. we can COULD a text/plain marshaller for String since we know how to represent it in JSON etc. It's not a 1:1 thing. Though this PR does restrict it to a 1:1 thing and blows up otherwise.